### PR TITLE
WIP: Travis CI Add Windows builds with Python 2.7 and 3.5

### PR DESCRIPTION
**Pull request by cclauss**
__Work in Progress...__  Do not merge.

As suggested at https://github.com/ray-project/ray/issues/631#issuecomment-477324678  @robertnishihara Your review please.

Allow the __Windows__ tests to fail until the scripts can be updated to deal with this OS.

### CODE REVIEW

**Comment by cclauss** [comment]
These errors have nothing to do with this PR's modifications.

**Comment by robertnishihara** [comment]
How about we don't allow failures, but also don't attempt the full Ray installation? That way, as we make more of the installation work (e.g., the first script to adapt would be `ci/travis/install-bazel.sh`), we can make sure it doesn't break.

**Comment by cclauss** [comment]
I am consistently amazed how folks resist __allow_failures__ given the logical explanation that Travis provides https://docs.travis-ci.com/user/customizing-the-build#rows-that-are-allowed-to-fail  Fixing code without tests to see where/why it is failing is sub-optimal.

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +22,20 @@ matrix:
       osx_image: xcode7
       env: PYTHON=3.5 PYTHONWARNINGS=ignore
 
+    - os: windows
+      language: shell  # 'language: python' is an error on Travis CI Windows
```
Comment (robertnishihara) [inline .travis.yml]: What about just omitting `language`? Then the `language: generic` from the top of the file will be used.

File: .travis.yml [hunk 2]
```diff
@@ -20,6 +22,20 @@ matrix:
...
+      before_install: choco install python2
```
Comment (robertnishihara) [inline .travis.yml]: I'm not familiar with `choco`, but what do you think about just getting Miniconda from https://docs.conda.io/en/latest/miniconda.html and installing that?

Comment (robertnishihara) [reply .travis.yml]: Actually, I think we should move this logic into https://github.com/ray-project/ray/blob/master/ci/travis/install-dependencies.sh, which is where we install Python for the other Travis builds.

File: .travis.yml (hunk 1 above)
Comment (cclauss) [reply .travis.yml]: Please read thru https://docs.travis-ci.com/user/reference/windows  __generic__ is not on the __supported-languages__ list and using it causes the build to fail.

Comment (cclauss) [reply .travis.yml]: https://chocolatey.org Is the tool that Travis builds their images with and it is bundled in the base image like homebrew is for their macOS images and apt is for their Linux images.  We can use https://chocolatey.org/search?q=Miniconda to install Miniconda.  I would prefer if we could first concentrate on getting the Ray test suite passing  on Windows.  After those tests pass, we can moving the build logic out of .travis.yml and into multiple files.

**Comment by cclauss** [comment]
@robertnishihara I have taken this as far as I can.  As discussed above __allow_failures: os: windows__ is __on__ because this PR does not get us all the way through the tests but at least it sets up __miniconde__ and __openssl__ and then installs __Python 2.7 and 3.5__ and then the __dependencies__.

Someone who has more __Ray__ skills than myself will need to pick up the thread once this PR has been merged.

**Pull request closed without merging**